Library/Formula/gobject-introspection.rb

```diff
@@ -27,13 +36,28 @@ def install
     ENV["GI_SCANNER_DISABLE_CACHE"] = "true"
     ENV.universal_binary if build.universal?
     inreplace "giscanner/transformer.py", "/usr/share", "#{HOMEBREW_PREFIX}/share"
-    inreplace "configure" do |s|
-      s.change_make_var! "GOBJECT_INTROSPECTION_LIBDIR", "#{HOMEBREW_PREFIX}/lib"
-    end
 
-    system "./configure", "--disable-dependency-tracking", "--prefix=#{prefix}"
-    system "make"
-    system "make", "install"
+    if build.head?
+      Language::Python.each_python(build) do |python, version|
+        sprintf "HEAD #{python} #{version}"
+        system "./autogen.sh"
+        inreplace "configure" do |s|
+          s.change_make_var! "GOBJECT_INTROSPECTION_LIBDIR", "#{HOMEBREW_PREFIX}/lib"
+        end
+        system "./configure", "--disable-dependency-tracking", "--prefix=#{prefix}", "PYTHON=python#{version}"
+        system "make"
+        system "make", "install"
+        system "make", "clean"
+      end
+    else
+      inreplace "configure" do |s|
+        s.change_make_var! "GOBJECT_INTROSPECTION_LIBDIR", "#{HOMEBREW_PREFIX}/lib"
+      end
+      system "./configure", "--disable-dependency-tracking", "--prefix=#{prefix}", "PYTHON=python#{version}"
+      system "make"
+      system "make", "install"
+      system "make", "clean"
```

Comment:
You shouldn't need to `make clean`. The build directory is disposed of after use.